public/js/addsales.js

```diff
@@ -0,0 +1,63 @@
+
+function init() {
+    var itemID = document.getElementById('ItemId');
+    var itemName = document.getElementById('ItemName');
+    var price = document.getElementById("Price");
+    var qty = document.getElementById("Quantity");
+    var date = document.getElementById("Date");
+    var statetxt = document.getElementById("statetxt");
+    var addSalesForm = document.getElementById("addSalesForm");
+    
+    
+    var idPattern = /^[a-zA-Z]+$/;
+    var namePattern = /^[A-Za-z ]+/;
+    var numberPattern = /^[0-9]+]/;
+    itemName.onkeypress = function() {  
+        if (!itemName.value.match(namePattern)) {
```

Comment:
Since the if ... else ... block is repeating many times. Can we refactor it down to 1 function with parameters are things-to-check, pattern, msg.